### PR TITLE
Merge improved date parser into existing bot

### PR DESCRIPTION
**Pull request by GellertPalfi**
Since the logic has been completely rewritten, i don't see any conflicts that would cause problems

### CODE REVIEW

**Review by mindedj — CHANGES_REQUESTED** [review]
It seems there are conflicts, you need to match the directory structure of refactor_nodejs (actions -> action_server, backend -> rasa_server)

File: rasa_server/.gitignore [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+dmypy.json
+
+# Pyre type checker
+.pyre/
+
+# PyCharm
+.idea/
+
+# dont upload models
+backend/models/
```
Comment (mindedj) [inline rasa_server/.gitignore:130]: rasa_server/models/

File: action_server/actions.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+    def run(self, dispatcher: CollectingDispatcher,
+            tracker: Tracker,
+            domain: Dict[Text, Any]) -> List[Dict[Text, Any]]:
+
+        time_table = get_timetable_in_discussion(tracker)
+        rule_blocks = RuleBlocks(tracker, time_table, dispatcher)
+        action_blocks = ActionBlocks(tracker, time_table, dispatcher)
+
+        if not rule_blocks.if_text_has_datetime():
```
Comment (mindedj) [inline action_server/actions.py:168]: With reversed if statement this block can get rid of 1 level of indentation

**Review by mindedj — COMMENTED** [review]
My suggestions are mostly for refactoring, I could not find anything that could be done 'easier'

**Review by mindedj — APPROVED** [review]

**Comment by nsomabalint** [comment]
As discussed we will address all comments by @mindedj in a future PR as we should do a larger refactor anyways. I'll merge this for now.

**Review by nsomabalint — APPROVED** [review]

**Pull request merged**